osf_tests/users/test_last_login_date.py

```diff
@@ -0,0 +1,44 @@
+import mock
+import pytz
+import pytest
+import itsdangerous
+from datetime import datetime
+
+from django.utils import timezone
+
+from website import settings
+
+from osf_tests.factories import (
+    AuthUserFactory,
+    SessionFactory
+)
+from tests.base import OsfTestCase
+
+@pytest.mark.django_db
+@pytest.mark.enable_enqueue_task
+class TestUserLastLoginDate(OsfTestCase):
+
+    def setUp(self):
+        super(TestUserLastLoginDate, self).setUp()
+
+        self.user = AuthUserFactory()
+
+        self.session = SessionFactory(
+            data={
+                'auth_user_id': self.user._id,
+                'auth_user_username': self.user.username
+            }
+        )
+        self.cookie = itsdangerous.Signer(settings.SECRET_KEY).sign(self.session._id).decode()
+
+    @mock.patch.object(timezone, 'now')
+    def test_date_last_login_updated(self, mock_time):
+        now = datetime(2018, 2, 4, tzinfo=pytz.utc)
+        mock_time.return_value = now
+        assert self.user.date_last_login is None
+
+        self.app.set_cookie(settings.COOKIE_NAME, self.cookie)
+        self.app.get(f'{settings.DOMAIN}{self.user._id}')  # user page will fail because not emberized
+
+        self.user.refresh_from_db()
+        assert self.user.date_last_login == now
```

Comment:
Should there be three test cases: `date_last_login` is `None`, `date_last_login` is less than the threshold, and `date_last_login` is greater than the threshold?